File: delivery/api/src/main/java/org/delivery/api/domain/token/model/TokenDto.java
package org.delivery.api.domain.token.model;

import java.time.LocalDateTime;

public record TokenDto(
	String token,
	LocalDateTime expiredAt
) {

	public static TokenDto of(final String token, final Long expirationTime) {
		final LocalDateTime expiredAt = LocalDateTime.now().plusHours(expirationTime);
		return new TokenDto(token, expiredAt);
	}

}

File: service/api-gateway/src/main/kotlin/org/delivery/apigateway/account/model/TokenDto.kt
package org.delivery.apigateway.account.model

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming
import java.time.LocalDateTime

@JsonNaming(value = PropertyNamingStrategies.SnakeCaseStrategy::class)
data class TokenDto(
    var token: String? = null,
    var expiredAt: LocalDateTime? = null
) {

    companion object {
        @JvmStatic
        fun of(token: String, expirationTime: Long?): TokenDto {
            val expiredAt = LocalDateTime.now().plusHours(expirationTime!!)
            return TokenDto(token, expiredAt)
        }
    }

}
